DatadogTrace/Sources/OpenTelemetry/OTelSpan.swift

```diff
@@ -109,54 +109,77 @@ internal class OTelSpan: OpenTelemetryApi.Span {
     }
 
     func end(time: Date) {
+        let semaphore = DispatchSemaphore(value: 0)
+        var ended = false
+        var tags: [String: String] = [:]
+
         queue.sync {
             guard isRecording else {
+                ended = true
+                semaphore.signal()
                 return
             }
             isRecording = false
+            tags = makeTags()
+            semaphore.signal()
+        }
+        semaphore.wait()
```

Comment:
question/ Why waiting on `semaphore` if the operation on `queue` is `sync {}` 🤔💭? Also, FYI the https://github.com/DataDog/dd-sdk-ios/pull/1615 is proposing to get rid of `tracer.queue`. Hope it will simplify threading here 🙏.

Reply:
err - semaphore not required, we are on sync. For some reason, I thought the calling thread will not be blocked when dispatching work on a Queue.